File: book/packt/MongoDB.for.Java.Developers/4735_Chapter 5/src/main/java/com/packtpub/mongo/chapter5/model/Customer.java
package com.packtpub.mongo.chapter5.model;

import javax.persistence.*;
import java.util.List;
import static javax.persistence.FetchType.EAGER;
import org.hibernate.annotations.GenericGenerator;

@Entity

public class Customer   {

    @Id 
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    private String id;

    private String country;
    private String name;

    @OneToMany(mappedBy = "customerFK", fetch = EAGER)
    private List<Item> items;

    public Customer() {     }

	
   // Getter/Setters omitted for brevity

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getCountry() {
        return country;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public List<Item> getItems() {
        return items;
    }

    public void setItems(List<Item> items) {
        this.items = items;
    }
}

File: book/packt/Scala.for.Java.Developers/Chapter 06/sbtjpasample/src/main/scala/se/sfjd/Customer.scala
package se.sfjd

import javax.persistence._
import scala.reflect.BeanProperty
 
@Entity
@Table(name = "customer")
class Customer(n: String) {
 
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "CUSTOMER_ID")
    @BeanProperty
    var id: Int = _
 
    @BeanProperty
    @Column(name = "NAME")
    var name: String = n
 
    def this() = this (null)
 
    override def toString = id + " = " + name
}